lib/html_renderer.rb

```diff
@@ -1,39 +1,79 @@
 module Gobstones
   class HtmlRenderer
+    def initialize(options)
+      @options = options
+    end
+
     def render_success(result)
-      @result = {
-        boards: prepare_boards([:initial, :final], result)
-      }
+      bind_result(
+        boards: prepare_boards([:initial, :final], result),
+        reason: prepare_reason(result[:reason])
+      )
+    end
 
-      bind
+    def render_error_check_final_board_failed_different_boards(result)
+      bind_result(
+        error: :check_final_board_failed_different_boards,
+        boards: prepare_boards([:initial, :expected, :actual], result)
+      )
     end
 
-    def render_error_check_final_board_failed(result)
-      @result = {
-        error: :check_final_board_failed,
-        boards: prepare_boards([:actual, :initial, :expected], result)
-      }
+    def render_error_check_final_board_failed_unexpected_boom(result)
+      bind_result(
+        error: :check_final_board_failed_unexpected_boom,
```

Comment:
Here you could avoid the parenthesis, by writing it in the same way you use `fail_with` in the checker